File: forge/src/main/java/dev/vanutp/tgbridge/forge/TelegramBridgeBootstrap.java
package dev.vanutp.tgbridge.forge;

import net.minecraftforge.fml.common.Mod;

@Mod(TelegramBridge.MOD_ID)
public final class TelegramBridgeBootstrap {
    private final TelegramBridge mod = new TelegramBridge();
    public TelegramBridgeBootstrap() {
        System.out.println("meow meow from forge");
        mod.init();
    }
}

File: forge/src/main/kotlin/dev/vanutp/tgbridge/forge/TelegramBridgeBootstrap.kt
package dev.vanutp.tgbridge.forge

import net.minecraft.client.MinecraftClient
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent
import net.minecraftforge.fml.event.lifecycle.FMLDedicatedServerSetupEvent
import org.apache.logging.log4j.Level
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import thedarkcolour.kotlinforforge.forge.MOD_BUS
import thedarkcolour.kotlinforforge.forge.runForDist

@Mod(TelegramBridge.MOD_ID)
object TelegramBridgeBootstrap {
    val mod = TelegramBridge()

    init {
        val obj = runForDist(
            clientTarget = {},
            serverTarget = {
                MOD_BUS.addListener(::onServerSetup)
            }
        )
    }

    private fun onServerSetup(event: FMLDedicatedServerSetupEvent) {
        mod.init()
    }
}
